File: ios/Wellvo/Models/WellvoError.swift
import Foundation

/// Centralized error type for all Wellvo services.
/// Provides user-facing localized descriptions for consistent error display.
enum WellvoError: LocalizedError {
    case network(Error)
    case auth(String)
    case notFound(String)
    case serverError(String)
    case offline
    case unknown(Error)

    var errorDescription: String? {
        switch self {
        case .network(let error):
            return "Network error: \(error.localizedDescription)"
        case .auth(let message):
            return message
        case .notFound(let resource):
            return "\(resource) not found."
        case .serverError(let message):
            return "Server error: \(message)"
        case .offline:
            return "You appear to be offline. Your action will be completed when you reconnect."
        case .unknown(let error):
            return error.localizedDescription
        }
    }
}

File: android/app/src/main/java/net/wellvo/android/network/WellvoError.kt
package net.wellvo.android.network

sealed class WellvoError : Exception() {
    data class Network(override val message: String = "Network error. Please check your connection.") : WellvoError()
    data class Auth(override val message: String = "Authentication failed. Please sign in again.") : WellvoError()
    data class NotFound(override val message: String = "The requested resource was not found.") : WellvoError()
    data class ServerError(override val message: String = "Server error. Please try again later.") : WellvoError()
    data class Offline(override val message: String = "You are offline. Your action will sync when reconnected.") : WellvoError()
    data class Unknown(override val message: String = "An unexpected error occurred.") : WellvoError()

    val localizedMessage: String
        get() = message ?: "An unexpected error occurred."
}
